Sources/General/BinaryMap.c

```diff
@@ -249,4 +280,136 @@ void BinaryMap_Or(BinaryMap *me, const BinaryMap* source)
+            if (BinaryMap_GetBit(me, x + 1, y - 1))
+                mask |= 128;
+            return mask;
+	}
+}
 
+void BinaryMap_CopyTo(const BinaryMap *me, BinaryMap *source)
+{
+    RectangleC defaultArea = RectangleC_ConstructFrom2Ints(me->width, me->height);
+    Point defaultPoint = Point_Construct(0, 0);
+    BinaryMap_CopyToArea(me, source, &defaultArea, &defaultPoint);
+}
+
+void BinaryMap_CopyToArea(const BinaryMap *me, const BinaryMap *source, const RectangleC *area, const Point *at)
+{
+    int shift = (int)((uint32_t)area->x & me->wordMask) - (int)((uint32_t)at->x & me->wordMask);
+    
+    UInt32Array1D vector = UInt32Array1D_Construct((area->width >> me->wordShift) + 2);
+    
+    for (int y = 0; y < area->height; y++)
+    {
+        Point areaOffset = Point_Construct(area->x, area->y + y);
+        Point atOffset = Point_Construct(at->x, at->y + y);
+        
+        LoadLine(source, &vector, &areaOffset, area->width);
+        if (shift >= 0)
+            ShiftLeft(me, &vector, shift);
+        else
+            ShiftRight(me, &vector, -shift);
+        
+        SaveLine(me, &vector, &atOffset, area->width);      
+    }
+    UInt32Array1D_Destruct(&vector);
+}
+
+void BinaryMap_AndNot(const BinaryMap *me, BinaryMap* source)
+{
+    Size area = BinaryMap_GetSize(source);
+    int vectorSize = (area.width >> me->wordShift) + 2;
+    
+    UInt32Array1D vector = UInt32Array1D_Construct(vectorSize);
+    UInt32Array1D srcVector = UInt32Array1D_Construct(vectorSize);
+
+    for (int y = 0; y<area.height; ++y) 
+    {
+        Point p = Point_Construct(0, y);
+        
+        LoadLine(me, &vector, &p, area.width);
+        LoadLine(source, &srcVector, &p, area.width);
+        
+        // The AND NOT
+        for (int i = 0; i<vectorSize; ++i) 
+        {
+            vector.data[i] &= ~srcVector.data[i];
+        }
+        
+        SaveLine(me, &vector, &p, area.width);
+    }
+
+    UInt32Array1D_Destruct(&vector);
+    UInt32Array1D_Destruct(&srcVector);
+}
+
+void BinaryMap_AndNotToArea(const BinaryMap *me, BinaryMap *source, const RectangleC *area, const Point *at)
+{
+	int shift = (int)((uint32_t)area->x & me->wordMask) - (int)((uint32_t)at->x & me->wordMask);
+	int vectorSize = (area->width >> me->wordShift) + 2;
+
+	UInt32Array1D vector = UInt32Array1D_Construct(vectorSize);
+	UInt32Array1D srcVector = UInt32Array1D_Construct(vectorSize);
+
+	for (int y = 0; y < area->height; y++)
+	{
+		Point atOffset = Point_Construct(at->x, at->y + y);
+		Point areaOffset = Point_Construct(area->x, area->y + y);
+		
+		LoadLine(me, &vector, &atOffset, area->width);
+		LoadLine(source, &srcVector, &areaOffset, area->width);
+
+		if (shift >= 0)
+		{
+			ShiftLeft(me, &srcVector, shift);
+		}
+		else
+		{
+			ShiftRight(me, &srcVector, -shift);
+		}
+
+		// The AND NOT
+		for (int i = 0; i < vector.size; ++i)
+		{
+			vector.data[i] &= ~srcVector.data[i];
+		}
+		SaveLine(me, &vector, &atOffset, area->width);
```

Comment:
This chunk of code (`// The AND NOT`) is duplicated in `BinaryMap_AndNot`. Same goes for the `And` logic around [line 251](https://github.com/SimPrints/libAFIS/pull/16/files#diff-fd5ba813ceaf98fcfa6cef584044ffe8R251) - it's in `BinaryMap_And` as well as `BinaryMap_AndArea`. Do we want to extract these out into their own functions to avoid duplication?